### PR TITLE
MPP-3441: add first_forwarded_email API endpoint

### PR DESCRIPTION
**Pull request by groovecoder**
This PR fixes #MPP-3441.
### Still TODO
* [x] Fix the `From` to be the mask email address

### How to test:
1. [Set up end-to-end local email](https://github.com/mozilla/fx-private-relay/blob/main/docs/end-to-end-local-email-dev.md) or push branch to dev server
2. Sign in and go to http://127.0.0.1:8000/api/v1/docs/#/first-forwarded-email
3. Expand the endpoint
4. Click "Try it out"
5. Enter a generated masks email as the value to "mask" in the request body.
6. Click "Execute"
   * [x] It should show a `201` response code
   * [x] It should send the first forwarded email

### Checklist
- [x] ~l10n changes have been submitted to the l10n repository, if any.~
- [ ] I've added a unit test to test for potential regressions of this bug.
- [x] ~I've added or updated relevant docs in the docs/ directory.~
- [x] ~All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).~
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: api/views/__init__.py [hunk 1]
```diff
@@ -343,6 +351,78 @@ def report_webcompat_issue(request):
     return response.Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
+class FirstForwardedEmailRateThrottle(throttling.UserRateThrottle):
+    rate = settings.FIRST_EMAIL_RATE_LIMIT
```
Comment (rafeerahman) [inline api/views/__init__.py:355]: Noting, when the user sends more then 5 requests in the frontned, need to handle it in the frontend @lloan .

File: api/views/__init__.py [hunk 2]
```diff
@@ -343,6 +351,78 @@ def report_webcompat_issue(request):
...
+                f"{mask} does not exist for user.", status=status.HTTP_404_NOT_FOUND
+            )
+        profile = user.profile
+        app_config = apps.get_app_config("emails")
+        assert isinstance(app_config, EmailsConfig)
+        ses_client = app_config.ses_client
+        assert ses_client
+        assert settings.RELAY_FROM_ADDRESS
+        with django_ftl.override(profile.language):
+            translated_subject = ftl_bundle.format("forwarded-email-hero-header")
```
Comment (rafeerahman) [inline api/views/__init__.py:395]: Question: What is this line doing and where is it being used?

File: api/views/__init__.py [hunk 3]
```diff
@@ -343,6 +351,78 @@ def report_webcompat_issue(request):
...
+    Note: `mask` value must be a `RelayAddress` that belongs to the authenticated user.
+    A `DomainAddress` will not work.
+    """
+    if not flag_is_active(request, "free_user_onboarding"):
+        # Return Permission Denied error
+        return response.Response(
+            {"detail": "Requires free_user_onboarding waffle flag."}, status=403
+        )
+    serializer = FirstForwardedEmailSerializer(data=request.data)
+    if serializer.is_valid():
```
Comment (rafeerahman) [inline api/views/__init__.py:378]: (nit) We could do `if not serializers.is_valid()` and `return response.Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)`, then unindent the line 378 to line 423

File: api/views/__init__.py [hunk 4]
```diff
@@ -343,6 +351,78 @@ def report_webcompat_issue(request):
...
+        mask = serializer.data.get("mask")
+        user = request.user
+        try:
+            address = _get_address(mask)
+            RelayAddress.objects.get(user=user, address=address)
+        except ObjectDoesNotExist:
+            return response.Response(
```
Comment (rafeerahman) [inline api/views/__init__.py:385]: (praise) Nice error handling

**Review by rafeerahman — APPROVED** [review]
Backend changes work as intended, tested on dev. 

Frontend content changes can be reviewed by @lloan once merged to #3962.

File: api/views/__init__.py (hunk 3 above)
Comment (groovecoder) [reply api/views/__init__.py]: I love early `return` coding! Nice catch!

**Pull request merged**